Enrichment analysis results, standalone GR
===
Hannum
------
- `09-20-2014_11.09.10AM_meQTL_vs_TFBS161` - meQTLs vs. 161 TFBSs
- `09-21-2014_08.56.45AM_meQTL50bp_vs_TFBS161` - meQTLs flanked +/-50bp vs. 161 TFBSs, using flanked background
- `09-21-2014_08.57.04AM_meQTL_vs_DGV` - vs. individual DGVs (Deletions, Gain/Loss, CNVs etc.)
- `09-21-2014_09.14.57AM_meQTL_vs_nestedRepeats` - vs. individual nested repeats (LINE, SINE etc.)
- `09-21-2014_10.43.49AM_meQTL_vs_Mist` - vs. evoFold, CpG, Vista enhancers, repeats, conservations 

Hannum_genes
------------
- `09-20-2014_10.12.50PM_TFBS161` - all genes vs. 161 TFBSs
- `09-20-2014_10.21.18PM_s6_vs_ENCODEmean3` - all expression-age genes (mmc4) vs. all ENCODE, mean threshold, 3 runs
- `09-20-2014_10.31.11PM_s7_vs_ENCODEmean3` - all expression-methylation-age genes (mmc5) vs. all ENCODE, mean threshold, 3 runs


Enrichment analysis results, web GR
-----------------------------------

[Tissue-specific aDMRs vs. chromatin states](http://genomerunner.org/result?id=khc6j46oojtys5rnq4u14cnhs3bvxsfn), saved in GR_mmc3_vs_chromStates.tar.gz

[Tissue-specific aDMRs vs. 

[Tissue-specific aDMRs vs. 

[Tissue-specific aDMRs vs. 

[Tissue-specific aDMRs vs. 


**Non-significant:**

Tissue-specific aDMRs vs. repeats, coriellCNVs, ncRNAs, gapLocations, altSplicing,


