---
title: "Analysis of age correlation with CpGs in naive CD4 T cells"
# author: "Mikhail G. Dozmorov"
# date: "March 22, 2016"
output: html_document
---

This code is based on carefully debugged `work_script.R`. It generates summary statistics about data processing steps, batch effect removal, and identification of the age-associated CpG probes.

```{r setup, echo=FALSE, message=FALSE, warning=FALSE}
# Set up the environment
library(knitr)
opts_chunk$set(cache.path='cache/', fig.path='img/', cache=F, tidy=T, fig.keep='high', echo=F, dpi=100, warnings=F, message=F, comment=NA, warning=F, results='as.is') #out.width=700, 
library(pander)
panderOptions('table.split.table', Inf)
set.seed(1)
```

```{r}
library(dplyr)
library(methylumi)
library(lumi)
library(xlsx)
library(ggplot2)
library(ggrepel)
library(wateRmelon)
library(betareg)
library(pander)
library(sva)
source("/Users/mikhail/Documents/Work/GenomeRunner/Aging/Illumina/450K_DataProcessing/SRC/filterXY.R")

# A function to pull out p-value of LM. https://stackoverflow.com/questions/5587676/pull-out-p-values-and-r-squared-from-a-linear-regression
lmp <- function (modelobject) {
  if (class(modelobject) != "lm") stop("Not an object of class 'lm' ")
  f <- summary(modelobject)$fstatistic
  p <- pf(f[1],f[2],f[3],lower.tail=F)
  attributes(p) <- NULL
  return(p)
}
```

```{r}
# Read in methylation values
# MethyFileName <- "data/subset_1000.csv.gz"
# MethyFileName <- "/Users/mikhail/Documents/Data/Work_OMRF/Amr/Control Naive CD4+ T cell Methylation/Control Naive CD4 Tcell Methylation Values.csv.gz"
# mldat <- methylumiR(filename = MethyFileName)
# save(list = "mldat", file = "data/mldat.rda")
load("/Users/mdozmorov/Documents/Data/Work_OMRF/Amr/Control Naive CD4+ T cell Methylation/mldat.rda")
```

```{r}
# Read in pheno data, and Match order of samples
PhenoFileName <- "data/Female SLE Naive CD4 T cell Control Info.xlsx"
samp <- read.xlsx2(file = PhenoFileName, sheetName = "Sheet1", stringsAsFactors = F)
samp$Age <- as.numeric(samp$Age) # Make age numeric
samp <- mutate(samp, names = paste(BeadChip.ID, Chip.Placement, sep = "_"))
samp <- samp[match(sampleNames(mldat), samp$names), ]
# Check names
all.equal(sampleNames(mldat), samp$names) # Should be true
rownames(samp) <- samp$names
# Add pheno data
pData(mldat) <- samp
```

# Population stats

```{r}
ind <- c("9611519008", "9611519009", "9974366123") # Batches with single samples, should be removed
samp <- samp[!(samp$BeadChip.ID %in% ind), ] # Remove them
table(samp$Race)
# Save the demographics data
unlink("Tables/Supplementary Table S1 patient info.xlsx")
write.xlsx2(samp[, c("BeadChip.ID", "Chip.Placement", "Race", "Age")], "Tables/Supplementary Table S1 patient info.xlsx", row.names = FALSE)
```

# Age distribution

```{r}
samp %>% ggplot(aes(x=Age)) + 
  geom_histogram(aes(y =..density..), col="red", fill="blue", alpha=0.5) +
  labs(title="Histogram for Age") +
  labs(x="Age", y="Percent") + 
  geom_density(col="black") +
  scale_x_continuous(breaks=seq(min(samp$Age), max(samp$Age), by = 2)) +
  scale_y_continuous(labels=percent) +
  theme(axis.text.x=element_text(size = 12), axis.text.y=element_text(size=12), legend.title=element_text(size=16, face = "bold"))
```


# QC check

First, we remove probes that have confounding NAs with the "BeadChip.ID" batch.

```{r}
print(paste("Total number of CpG probes to start with:", nrow(mldat)))
# Batches
batch.BeadChip.ID <- table(pData(mldat)$BeadChip.ID) %>% sort # Summary how many samples per batch
batch.BeadChip.ID.one <- names(batch.BeadChip.ID)[ batch.BeadChip.ID == 1 ] # Sample names associated with one-sample batch
batch.BeadChip.ID.three <- names(batch.BeadChip.ID)[ batch.BeadChip.ID == 3 ] # Sample names associated with three-sample batch
batch.BeadChip.ID.four <- names(batch.BeadChip.ID)[ batch.BeadChip.ID == 4 ] # Etc.
batch.BeadChip.ID.six <- names(batch.BeadChip.ID)[ batch.BeadChip.ID == 6 ]
batch.BeadChip.ID.twelve <- names(batch.BeadChip.ID)[ batch.BeadChip.ID == 12 ]

# Removing CpG cites having NAs in batches
ind.one <- pData(mldat)$BeadChip.ID %in% batch.BeadChip.ID.one # TRUE/FALSE vector of samples associated with one-sample batch
ind.three <- pData(mldat)$BeadChip.ID %in% batch.BeadChip.ID.three # TRUE/FALSE vector of samples associated with three-sample batch
ind.four <- pData(mldat)$BeadChip.ID %in% batch.BeadChip.ID.four # Etc
ind.six <- pData(mldat)$BeadChip.ID %in% batch.BeadChip.ID.six
ind.twelve <- pData(mldat)$BeadChip.ID %in% batch.BeadChip.ID.twelve
ind.exclude <- vector(mode = "numeric") # Row numbers to exclude
for (i in 1:nrow(mldat)) {
  # If a row (CpG probe) contains NAs in samples associated with one- or three- or ... -sample batches, keep its index 
  if (any(sapply(batch.BeadChip.ID.one, function(x) sum(is.na(betas(mldat)[i, ind.one])) == 1 )) |
      any(sapply(batch.BeadChip.ID.three, function(x) sum(is.na(betas(mldat)[i, ind.three])) == 3 )) |
      any(sapply(batch.BeadChip.ID.three, function(x) sum(is.na(betas(mldat)[i, ind.four])) == 4 )) |
      any(sapply(batch.BeadChip.ID.three, function(x) sum(is.na(betas(mldat)[i, ind.six])) == 6 )) |
      any(sapply(batch.BeadChip.ID.three, function(x) sum(is.na(betas(mldat)[i, ind.twelve])) == 12 ))
      ) {
    ind.exclude <- c(ind.exclude, i)
  }
}
mldat <- mldat[-ind.exclude, ] # Remove offending rows
print(paste("Total number of CpG probes after removing CpG probes with NAs confounding batch:", nrow(mldat)))
```

Then, we identify which batch effect contributes most to data variability, using Principle Aomponents Analysis

```{r}
# PCA: Check for batch effects. Select one batch, to color points by its assignment
# batch <- "Race"
batch <- "BeadChip.ID"
# batch <- "Chip.Placement"
pca <- betas(mldat)[ complete.cases(betas(mldat)) & apply(betas(mldat), 1, sd) != 0, ] %>% scale %>% t %>% prcomp
data.frame(summary(pca)$importance)[, 1:5] %>% pander # Percent of variance explained
# What correlates with the first component
pca.lm <- lm( as.numeric(PC1) ~ factor(BeadChip.ID), data = cbind(pData(mldat), pca$x))
print(paste("BeadChip.ID accounts for", signif(summary(pca.lm)$adj.r.squared, 5), "variability explained by the first principle component, p-value", signif(lmp(pca.lm), 5)))
pca.lm <- lm( as.numeric(PC1) ~ factor(Chip.Placement), data = cbind(pData(mldat), pca$x))
print(paste("Chip.Placement accounts for", signif(summary(pca.lm)$adj.r.squared, 5), "variability explained by the first principle component, p-value", signif(lmp(pca.lm), 5)))
pca.lm <- lm( as.numeric(PC1) ~ factor(Race), data = cbind(pData(mldat), pca$x))
print(paste("Race accounts for", signif(summary(pca.lm)$adj.r.squared, 5), "variability explained by the first principle component, p-value", signif(lmp(pca.lm), 5)))

pt <- ggplot(data = data.frame(pca$x, pData(mldat), samples = pData(mldat)$Sample.ID, stringsAsFactors = F), 
             aes(x = as.numeric(PC1), y = as.numeric(PC2), label = eval(parse(text = batch)))) +
  theme(plot.title = element_text(lineheight = 0.8, face="bold")) +
  ggtitle(paste("Batch coloring:", batch)) +
  geom_point(aes(color = eval(parse(text = batch))), size = 3) +
  geom_text_repel(colour = "black", size = 3) +
  geom_hline(yintercept = 0, colour = "gray65") +
  geom_vline(xintercept = 0, colour = "gray65") +
  labs(color = batch) +
  scale_x_continuous(name = "PC1 (10.48%)") +
  scale_y_continuous(name = "PC2 (7.95%)")
plot(pt)

ggsave(filename = "Figures/Figure_PCA_batch.pdf", plot = pt, height = 8, width = 11)
```

# Normalization

Before normalization we filter non-specific, polymorphic probes, chromosome Y

```{r}
# # Filtering
print(paste("Number of probes before filtering:", nrow(mldat)))
# # Exclude CpG probes with SNPs
# snpsites <- read.table("/Users/mikhail/Documents/Work/GenomeRunner/Aging/Illumina/snpsites.txt.gz", sep="\t", stringsAsFactors = FALSE)
# ind <- which(is.element(featureNames(mldat), snpsites$V1))
# mldat <- mldat[-ind, ]
# Exclude non-specific probes
annot.nonspecific <- openxlsx::read.xlsx("/Users/mikhail/Documents/Work/GenomeRunner/Aging/Illumina/48639-non-specific-probes-Illumina450k.xlsx", sheet = 1, cols = 1)
ind <- which(is.element(featureNames(mldat), annot.nonspecific$TargetID))
mldat <- mldat[-ind, ]
print(paste("After removing", length(ind), "non-specific probes,", nrow(mldat), "probes remain."))
# Exclude polymorphic probes
annot.polymorphic <- openxlsx::read.xlsx("/Users/mikhail/Documents/Work/GenomeRunner/Aging/Illumina/48640-polymorphic-CpGs-Illumina450k.xlsx", sheet = 1, cols = 1)
ind <- which(is.element(featureNames(mldat), annot.polymorphic$PROBE))
mldat <- mldat[-ind, ]
print(paste("After removing", length(ind), " polymorphic probes,", nrow(mldat), "probes remain."))
# # Filter probes showing detection p-value > 0.05 in at least 10% of the samples
# mldat <- detectionPval.filter2(mldat, detectionPval.threshold=0.05, detectionPval.perc.threshold=10, projectName = NULL, PATH="./")
# print(paste("After removing probes showing detection p-value > 0.05 in at least 10% of the samples,", nrow(mldat), "probes remain."))
# Filter chrY probes
mldat <- filterXY(mldat, chr_remove = c("Y"))
print(paste("After removing chrY probes,", nrow(mldat), "probes remain."))
# Remove probes with median betas = 0% or 100% across all samples
ind <- rowMedians(betas(mldat), na.rm = T)
mldat <- mldat[ ind != 1 | ind != 0, ]
print(paste("After removing probes with median betas = 0% or 100% across all samples,", nrow(mldat), "probes remain."))
```

The data is filtered/normalized using best practices of the `wateRmelon` package, "dasen" followed "BIMQ" normalization.

```{r eval=FALSE}
# Normalization 
# quality filter using default thresholds
melon.pf <- pfilter(mldat)
# preprocess using our best method
melon.dasen.pf <- dasen(melon.pf)
melon.dasen.pf <- BMIQ(melon.dasen.pf, nfit=100)
```

# Batch effect adjustment

We adjust for the "BeadChip.ID" batch effect, using ComBat function. As a sanity check, we compare correlation between raw and adjusted data - should be closer to 1. 

```{r eval=FALSE}
batch <- pData(melon.dasen.pf)$BeadChip.ID
modcombat <- model.matrix(~1 , data=pData(melon.dasen.pf)) # + BeadChip.ID + Chip.Placement
combat_edata <- sva::ComBat(betas(melon.dasen.pf), batch = batch, mod = modcombat, par.prior = TRUE, prior.plots = FALSE)
save(list = c("combat_edata"), file = "data/combat_edata_parpriorTrue.rda")
```

```{r eval=FALSE}
# Sanity check: Comparing correlations before/after adjustments
cor.raw <- cor(pData(melon.dasen.pf)$Age, t(betas(melon.dasen.pf))) # Correlation with age before any adjustments
cor.adj <- cor(pData(melon.dasen.pf)$Age, t(combat_edata)) # Correlation with age after batch adjustment
cor(as.numeric(cor.raw), as.numeric(cor.adj), use = "complete.obs") # How raw and adjusted vectors of correlation coefficients compare?
```

# PCA after batch removal

```{r}
# PCA: Check for batch effects. Select one batch, to color points by its assignment
# batch <- "Race"
batch <- "BeadChip.ID"
# batch <- "Chip.Placement"
pca <- combat_edata[ complete.cases(combat_edata) & apply(combat_edata, 1, sd) != 0, ] %>% scale %>% t %>% prcomp
data.frame(summary(pca)$importance)[, 1:5] %>% pander # Percent of variance explained
# What correlates with the first component
pca.lm <- lm( as.numeric(PC1) ~ factor(BeadChip.ID), data = cbind(annot, pca$x))
print(paste("BeadChip.ID accounts for", signif(summary(pca.lm)$adj.r.squared, 5), "variability explained by the first principle component, p-value", signif(lmp(pca.lm), 5)))
pca.lm <- lm( as.numeric(PC1) ~ factor(Chip.Placement), data = cbind(annot, pca$x))
print(paste("Chip.Placement accounts for", signif(summary(pca.lm)$adj.r.squared, 5), "variability explained by the first principle component, p-value", signif(lmp(pca.lm), 5)))
pca.lm <- lm( as.numeric(PC1) ~ factor(Race), data = cbind(annot, pca$x))
print(paste("Race accounts for", signif(summary(pca.lm)$adj.r.squared, 5), "variability explained by the first principle component, p-value", signif(lmp(pca.lm), 5)))

pt <- ggplot(data = data.frame(pca$x, annot, samples = annot$Sample.ID, stringsAsFactors = F), 
             aes(x = as.numeric(PC1), y = as.numeric(PC2), label = eval(parse(text = batch)))) +
  theme(plot.title = element_text(lineheight = 0.8, face="bold")) +
  ggtitle(paste("Batch coloring:", batch)) +
  geom_point(aes(color = eval(parse(text = batch))), size = 3) +
  geom_text_repel(colour = "black", size = 3) +
  geom_hline(yintercept = 0, colour = "gray65") +
  geom_vline(xintercept = 0, colour = "gray65") +
  labs(color = batch) +
  scale_x_continuous(name = "PC1 (12.46%)") +
  scale_y_continuous(name = "PC2 (4.39%)")
plot(pt)

ggsave(filename = "Figures/Figure_PCA_nobatch.pdf", plot = pt, height = 8, width = 11)
```

# Correlation analysis

We perform standard linear regression, `betareg` linear regression, and Pearson correlation-based analysis. Results are in "results/correlation_results.txt"

```{r eval=FALSE}
# Correlation analysis
mldat.final = new(Class = "MethyLumiSet")
betas(mldat.final) <- combat_edata
pData(mldat.final) <- pData(melon.dasen.pf)

# Allocate storage
coeffs <- vector(mode = "list", length = nrow(mldat.final)) # Store coefficients for regular regression
adj.r.squared <- vector(mode = "list", length = nrow(mldat.final)) # R2 for regular regression
lmpval <- vector(mode = "list", length = nrow(mldat.final)) # p-value for regular regression
betareg.coeffs <- vector(mode = "list", length = nrow(mldat.final)) # Store coefficients for betareg regression
betareg.adj.r.squared <- vector(mode = "list", length = nrow(mldat.final)) # R2 for betareg regression
correl <- vector(mode = "list", length = nrow(mldat.final)) # Pearson correlation coefficients

# Test each row (CpG expression profile) for association with Age
for (i in 1:nrow(mldat.final)) {
#  print(i)
  y <- betas(mldat.final)[i, ] # Vector of beta values to test
  fit <- lm(y ~ Age, data = pData(mldat.final), na.action=na.omit) # Age only
  #  fit2 <- lm(betas(mldat.lumi.quantile.final)[i, ] ~ Age + Race + BeadChip.ID + Chip.Placement, data = pData(mldat.lumi.quantile.final)) # Age plus batches
  coeffs[[i]] <- c(summary(fit)$coefficients[2, "Estimate"], summary(fit)$coefficients[2, "Pr(>|t|)"]) #, summary(fit2)$coefficients[2, "Estimate"], summary(fit2)$coefficients[2, "Pr(>|t|)"]) # The first coefficient, ignore intercept
  adj.r.squared[[i]] <- c(summary(fit)$adj.r.squared) #, summary(fit2)$adj.r.squared)
  lmpval[[i]] <- c(lmp(fit)) #, lmp(fit2))
  # betareg regression
  y[y > 1] <- 0.9999999999999999 # Set large outliers to maximum of (0, 1) range
  y[y < 0] <- 0.0000000000000001 # Set small outliers to minimum of (0, 1) range
  betareg.fit <- betareg(y ~ Age, data = pData(mldat.final), na.action=na.omit)
  betareg.coeffs[[i]] <- c(summary(betareg.fit)$coefficients$mean[2, "Estimate"], summary(betareg.fit)$coefficients$mean[2, "Pr(>|z|)"]) #, summary(fit2)$coefficients[2, "Estimate"], summary(fit2)$coefficients[2, "Pr(>|t|)"]) # The first coefficient, ignore intercept
  betareg.adj.r.squared[[i]] <- c(summary(betareg.fit)$pseudo.r.squared) #, summary(fit2)$adj.r.squared)
  # Correlation
  corr <- Hmisc::rcorr(betas(mldat.final)[i, ], Biobase::pData(mldat.final)$Age)
  correl[[i]] <- c(corr$r[1, 2], corr$P[1, 2])
  if (i %% 100000 == 0) { save(list = c("fit", "coeffs", "adj.r.squared", "lmpval", "correl"), file = paste0("results/lm_results_", i, ".rda")) }
}

# Combine results
results <- cbind(do.call(cbind, lapply(coeffs, data.frame, stringsAsFactors=F)) %>% t,
                 do.call(cbind, lapply(adj.r.squared, data.frame, stringsAsFactors=F)) %>% t,
                 do.call(cbind, lapply(lmpval, data.frame, stringsAsFactors=F)) %>% t,
                 do.call(cbind, lapply(betareg.coeffs, data.frame, stringsAsFactors=F)) %>% t,
                 do.call(cbind, lapply(betareg.adj.r.squared, data.frame, stringsAsFactors=F)) %>% t,
                 do.call(cbind, lapply(correl, data.frame, stringsAsFactors=F)) %>% t)
rownames(results) <- rownames(betas(mldat.final))
colnames(results) <- c("coeff.lm", "pval.lm", "r2.lm", "lmpval", "coeff.br", "pval.br", "r2.br", "corr.rcorr", "pval.rcorr")
write.table(results, "results/correlation_results.txt", sep = "\t", quote = FALSE, col.names = NA)
```

```{r session_info}
diagnostics <- devtools::session_info()
platform <- data.frame(diagnostics$platform %>% unlist, stringsAsFactors = FALSE) 
colnames(platform) <- c("description") 
pander(platform)

packages <- as.data.frame(diagnostics$packages) 
pander(packages[ packages$`*` == "*", ]) 
```
