---
title: "Overlap of age-associated genes with gene lists of interest"
# author: "Mikhail G. Dozmorov"
# date: "April 17, 2016"
output: html_document
---

```{r setup, echo=FALSE, include=FALSE, cache=FALSE}
# Set up the environment
library(knitr) 
opts_chunk$set(cache.path='cache/', fig.path='img/', cache=F, tidy=T, fig.keep='high', dpi=1000,  echo=F, autodep=T, width=200, warning=FALSE, comment=NA)
options(replace.assign=TRUE, width=500)
tidy.opts=list(width.cutoff=200)
suppressMessages(library(pander))
panderOptions('table.split.table', Inf)
set.seed(1)
library(dplyr)
```

Prepare list of all genes ...

```{r}
# Read in 450K annotation
ANNOT <- "/Users/mikhail/Documents/Work/GenomeRunner/Aging/Illumina/GPL13534-11288.txt.gz"
annot.cg <- read.table("/Users/mikhail/Documents/Work/GenomeRunner/Aging/Illumina/GPL13534-11288.txt.gz", sep = "\t", skip=37, header=T, stringsAsFactors = FALSE, fill = T)
# Select genes only
all_genes_on_450K <- annot.cg$UCSC_RefGene_Name[ annot.cg$UCSC_RefGene_Name != "" ]
all_genes_on_450K <- sapply(all_genes_on_450K, function(x) strsplit(x, ";")) %>% unlist %>% sort %>% unique %>% as.character
```

Prepare gene lists ...

- `gene_list_1` - ADME core genes
- `gene_list_2` - ADME extended genes
- `gene_list_3` - ADME related genes
- `gene_list_4` - Coagulation
- `gene_list_5` - Heparan Sulfate Biosynthesis
- `gene_list_6` - N-Linked Glycosylation
- `gene_list_7` - O-Linked Glycosylation
- `gene_list_8` - Glycosylation

```{r}
gene_list <- list(
  gene_list_1 = readLines("/Users/mikhail/Documents/Work/GenomeRunner/gwas2bed/genes/data/ADME_core.txt"),
  gene_list_2 = readLines("/Users/mikhail/Documents/Work/GenomeRunner/gwas2bed/genes/data/ADME_extended.txt"),
  gene_list_3 = readLines("/Users/mikhail/Documents/Work/GenomeRunner/gwas2bed/genes/data/ADME_related.txt"),
  gene_list_4 = as.character(unlist(openxlsx::read.xlsx("/Users/mikhail/Documents/Work/GenomeRunner/gwas2bed/genes/data/Genes_Glycosylation and Coagulation Genes.xlsx", sheet = 1, startRow = 4, colNames = FALSE, cols = 1))),
  gene_list_5 = as.character(unlist(openxlsx::read.xlsx("/Users/mikhail/Documents/Work/GenomeRunner/gwas2bed/genes/data/Genes_Glycosylation and Coagulation Genes.xlsx", sheet = 1, startRow = 4, colNames = FALSE, cols = 2))),
  gene_list_6 = as.character(unlist(openxlsx::read.xlsx("/Users/mikhail/Documents/Work/GenomeRunner/gwas2bed/genes/data/Genes_Glycosylation and Coagulation Genes.xlsx", sheet = 1, startRow = 4, colNames = FALSE, cols = 3))),
  gene_list_7 = as.character(unlist(openxlsx::read.xlsx("/Users/mikhail/Documents/Work/GenomeRunner/gwas2bed/genes/data/Genes_Glycosylation and Coagulation Genes.xlsx", sheet = 1, startRow = 4, colNames = FALSE, cols = 4))),
  gene_list_8 = as.character(unlist(openxlsx::read.xlsx("/Users/mikhail/Documents/Work/GenomeRunner/gwas2bed/genes/data/Genes_Glycosylation and Coagulation Genes.xlsx", sheet = 1, startRow = 4, colNames = FALSE, cols = 5)))
)
```

Prepare enrichment functions ...

```{r}
# Enrichment of genes in an gene_list list
enrichment <- function(genes_of_interest = genes_of_interest, all_genes_on_450K = all_genes_on_450K, gene_list = gene_list[[ "gene_list_1" ]]) {
  # Prepare 2x2 contingency table
  genes_of_interest.AND.gene_list <- length(intersect(genes_of_interest, gene_list))
  genes_of_interest.NOT.gene_list <- length(genes_of_interest) - genes_of_interest.AND.gene_list
  NOT.genes_of_interest.AND.gene_list <- setdiff(all_genes_on_450K, genes_of_interest) %>% intersect(., gene_list) %>% length
  NOT.genes_of_interest.NOT.gene_list <- length(all_genes_on_450K) - NOT.genes_of_interest.AND.gene_list - length(genes_of_interest)
  matrix.2x2 <- matrix(c(genes_of_interest.AND.gene_list, genes_of_interest.NOT.gene_list, NOT.genes_of_interest.AND.gene_list, NOT.genes_of_interest.NOT.gene_list), nrow = 2)
  # Precaution against zeros
  if (genes_of_interest.AND.gene_list == 0) {
    return("Nothing significant")
  }
  return(fisher.test(matrix.2x2))
}

# Wrapper to print the results
enrichment.all <- function(genes_of_interest = genes_of_interest, genes_of_interest_name = "", all_genes_on_450K = all_genes_on_450K, gene_list = gene_list[[ "gene_list_1" ]], gene_list_name = names(gene_list)[1]) {
  res <- enrichment(genes_of_interest, all_genes_on_450K, gene_list = gene_list)
  if (class(res) == "character") {
    # print(paste("Age genes do not contain", gene_list_name, "genes"))
  } else {
    genes_intersecting_gene_list <- intersect(genes_of_interest, gene_list)
    print(paste(length(genes_of_interest), genes_of_interest_name, "genes contain", length(genes_intersecting_gene_list), "genes in", gene_list_name, ifelse(res$estimate > 0, ", enriched", ", depleted"), "at Fisher's exact p-value", formatC(res$p.value, digits = 3, format = "e"), ifelse(res$p.value > 0.05, ", not significant", "") ))
    print(genes_intersecting_gene_list)
    if (any(grepl("CYP", genes_intersecting_gene_list))) {
      print(paste("CYP genes found:", paste(genes_intersecting_gene_list[grepl("CYP", genes_intersecting_gene_list)], collapse = ",") ))
    }
  }
}
```

### `Pair-wise enrichment of lists of genes of interest`

```{r}
pair_wise_combinations <- combn(names(gene_list), m = 2)

apply(pair_wise_combinations, 2, function(x) {
  results <- enrichment.all(gene_list[[ x[1] ]], x[1], all_genes_on_450K, gene_list[[ x[2] ]], x[2])
})
```


### `Alisch`

Alisch R. et. al. ["Age-associated DNA methylation in pediatric populations"](http://genome.cshlp.org/content/22/4/623.full)

```{r}
genes <- readLines("../data/Alisch/genes.txt")
genes_name <- "Alisch"
for (name in names(gene_list)) {
  enrichment.all(genes, genes_name, all_genes_on_450K, gene_list[[name]], name)
}
```

### `Fernandez`

Fernández AF, et.al. ["H3K4me1 marks DNA regions hypomethylated during aging in human stem and differentiated cells"](http://genome.cshlp.org/content/25/1/27/suppl/DC1) Genome Res 2015

Currently skipped, too many genes

**Mesenchymal Stem Cells**

```{r eval = FALSE}
genes <- readLines("../data/Fernandez/genes_MSC.txt")
for (name in names(gene_list)) {
  enrichment.all(genes, all_genes_on_450K, gene_list[[name]], name)
}
```

**Blood**

```{r eval = FALSE}
genes <- readLines("../data/Fernandez/genes_blood.txt")
for (name in names(gene_list)) {
  enrichment.all(genes, all_genes_on_450K, gene_list[[name]], name)
}
```

### `Florath`

Florath et al., [“Cross-Sectional and Longitudinal Changes in DNA Methylation with Age.”](http://hmg.oxfordjournals.org/content/23/5/1186.long)

```{r}
genes <- readLines("../data/Florath/genes.txt")
genes_name <- "Florath"
for (name in names(gene_list)) {
  enrichment.all(genes, genes_name, all_genes_on_450K, gene_list[[name]], name)
}
```

### `genes/deMagalhaes`

De Magalhães, J. P., et.al. ["Meta-analysis of age-related gene expression profiles identifies common signatures of aging"](http://bioinformatics.oxfordjournals.org/content/25/7/875.short)

```{r}
genes <- readLines("../data/genes/deMagalhaes/genes.txt")
genes_name <- "deMagalhaes"
for (name in names(gene_list)) {
  enrichment.all(genes, genes_name, all_genes_on_450K, gene_list[[name]], name)
}
```

## `genes/GenAge`

Data from [Human Ageing Genomic Resources](http://genomics.senescence.info/download.html)

```{r}
genes <- readLines("../data/genes/GenAge/genes.txt")
genes_name <- "GenAge"
for (name in names(gene_list)) {
  enrichment.all(genes, genes_name, all_genes_on_450K, gene_list[[name]], name)
}
```

### `genes/JenAge`

Data from [JenAge/AgeFactDB](http://agefactdb.jenage.de/cgi-bin/jaDB.cgi?VIEW=download).

```{r}
genes <- readLines("../data/genes/JenAge/genes.txt")
genes_name <- "JenAge"
for (name in names(gene_list)) {
  enrichment.all(genes, genes_name, all_genes_on_450K, gene_list[[name]], name)
}
```

### `genes/LongevityMap`

Data from [Human Ageing Genomic Resources](http://genomics.senescence.info/download.html)

```{r}
genes <- readLines("../data/genes/LongevityMap/genes.txt")
genes_name <- "LongevityMap"
for (name in names(gene_list)) {
  enrichment.all(genes, genes_name, all_genes_on_450K, gene_list[[name]], name)
}
```

### `genes/Peters`

Peters MJ, et. al. ["The transcriptional landscape of age in human peripheral blood"](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4639797). Nat Commun 2015

**Blood**

```{r}
genes <- readLines("../data/genes/Peters/genes_blood.txt")
genes_name <- "Peters, blood"
for (name in names(gene_list)) {
  enrichment.all(genes, genes_name, all_genes_on_450K, gene_list[[name]], name)
}
```

**CD4+**

```{r}
genes <- readLines("../data/genes/Peters/genes_CD4+.txt")
genes_name <- "Peters, CD4+"
for (name in names(gene_list)) {
  enrichment.all(genes, genes_name, all_genes_on_450K, gene_list[[name]], name)
}
```

**CD8+**

```{r}
genes <- readLines("../data/genes/Peters/genes_CD8+.txt")
genes_name <- "Peters, CD8+"
for (name in names(gene_list)) {
  enrichment.all(genes, genes_name, all_genes_on_450K, gene_list[[name]], name)
}
```

**Monocytes**

```{r}
genes <- readLines("../data/genes/Peters/genes_monocytes.txt")
genes_name <- "Peters, Monocytes"
for (name in names(gene_list)) {
  enrichment.all(genes, genes_name, all_genes_on_450K, gene_list[[name]], name)
}
```

**Lymphocytes**

```{r}
genes <- readLines("../data/genes/Peters/genes_lymphocytes.txt")
genes_name <- "Peters, Lymphocytes"
for (name in names(gene_list)) {
  enrichment.all(genes, genes_name, all_genes_on_450K, gene_list[[name]], name)
}
```

**LCLs**

```{r}
genes <- readLines("../data/genes/Peters/genes_LCLs.txt")
genes_name <- "Peters, LCLs"
for (name in names(gene_list)) {
  enrichment.all(genes, genes_name, all_genes_on_450K, gene_list[[name]], name)
}
```

**Cerebellum**

```{r}
genes <- readLines("../data/genes/Peters/genes_cerebellum.txt")
genes_name <- "Peters, Cerebellum"
for (name in names(gene_list)) {
  enrichment.all(genes, genes_name, all_genes_on_450K, gene_list[[name]], name)
}
```

**Frontal Cortex**

```{r}
genes <- readLines("../data/genes/Peters/genes_frontalcortex.txt")
genes_name <- "Peters, Frontal Cortex"
for (name in names(gene_list)) {
  enrichment.all(genes, genes_name, all_genes_on_450K, gene_list[[name]], name)
}
```

### `genes/Sood`

Sood et al., [“A Novel Multi-Tissue RNA Diagnostic of Healthy Ageing Relates to Cognitive Health Status.”](http://genomebiology.biomedcentral.com/articles/10.1186/s13059-015-0750-x)

```{r}
genes <- readLines("../data/genes/Sood/genes.txt")
genes_name <- "Sood"
for (name in names(gene_list)) {
  enrichment.all(genes, genes_name, all_genes_on_450K, gene_list[[name]], name)
}
```

### `Hannum`

Hannum G. et. al. ["Genome-wide methylation profiles reveal quantitative views of human aging rates"](http://www.sciencedirect.com/science/article/pii/S1097276512008933).

**All - The 89 age-associated methylation regions identified using all data**

```{r}
genes <- readLines("../data/Hannum/genes_all.txt")
genes_name <- "Hannum, all"
for (name in names(gene_list)) {
  enrichment.all(genes, genes_name, all_genes_on_450K, gene_list[[name]], name)
}
```

**Primary - The 71 age-associated methylation regions included in the primary aging model**

```{r}
genes <- readLines("../data/Hannum/genes_primary.txt")
genes_name <- "Hannum, primary"
for (name in names(gene_list)) {
  enrichment.all(genes, genes_name, all_genes_on_450K, gene_list[[name]], name)
}
```

**Associated**

```{r}
genes <- readLines("../data/Hannum/genes_associated.txt")
genes_name <- "Hannum, associated"
for (name in names(gene_list)) {
  enrichment.all(genes, genes_name, all_genes_on_450K, gene_list[[name]], name)
}
```

**Breast**

```{r}
genes <- readLines("../data/Hannum/genes_breast.txt")
genes_name <- "Hannum, breast"
for (name in names(gene_list)) {
  enrichment.all(genes, genes_name, all_genes_on_450K, gene_list[[name]], name)
}
```

**Kidney**

```{r}
genes <- readLines("../data/Hannum/genes_kidney.txt")
genes_name <- "Hannum, kidney"
for (name in names(gene_list)) {
  enrichment.all(genes, genes_name, all_genes_on_450K, gene_list[[name]], name)
}
```

**Lung**

```{r}
genes <- readLines("../data/Hannum/genes_lung.txt")
genes_name <- "Hannum, lung"
for (name in names(gene_list)) {
  enrichment.all(genes, genes_name, all_genes_on_450K, gene_list[[name]], name)
}
```

### `Heyn`

Heyn, H., et.al. ["Distinct DNA methylomes of newborns and centenarians"](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3387108/)

```{r}
genes <- readLines("../data/Heyn/genes.txt")
genes_name <- "Heyn"
for (name in names(gene_list)) {
  enrichment.all(genes, genes_name, all_genes_on_450K, gene_list[[name]], name)
}
```

### `Horvath`

Horvath S: "[DNA methylation age of human tissues and cell types](http://genomebiology.biomedcentral.com/articles/10.1186/gb-2013-14-10-r115#MOESM1)". Genome Biol 2013

```{r}
genes <- readLines("../data/Horvath/genes.txt")
genes_name <- "Horvath"
for (name in names(gene_list)) {
  enrichment.all(genes, genes_name, all_genes_on_450K, gene_list[[name]], name)
}
```

### `Marttila`

Marttila S, et.al. "[Ageing-associated changes in the human DNA methylome: genomic locations and effects on gene expression](http://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-015-1381-z)". BMC Genomics 2015

```{r}
genes <- readLines("../data/Marttila/genes.txt")
genes_name <- "Marttila"
for (name in names(gene_list)) {
  enrichment.all(genes, genes_name, all_genes_on_450K, gene_list[[name]], name)
}
```

### `Rakyan`

Rakyan V. et. al. ["Human aging-associated DNA hypermethylation occurs preferentially at bivalent chromatin domains"](http://genome.cshlp.org/content/early/2010/03/09/gr.103101.109)

```{r}
genes <- readLines("../data/Rakyan/genes.txt")
genes_name <- "Rakyan"
for (name in names(gene_list)) {
  enrichment.all(genes, genes_name, all_genes_on_450K, gene_list[[name]], name)
}
```

### `Reynolds`

Reynolds, L. M. et. al. ["Age-related variations in the methylome associated with gene expression in human monocytes and T cells."](http://www.nature.com/ncomms/2014/141118/ncomms6366/full/ncomms6366.html#supplementary-information)

**CD4+**

```{r}
genes <- readLines("../data/Reynolds/genes_CD4+.txt")
genes_name <- "Reynolds, CD4+"
for (name in names(gene_list)) {
  enrichment.all(genes, genes_name, all_genes_on_450K, gene_list[[name]], name)
}
```

**Monocytes**

```{r}
genes <- readLines("../data/Reynolds/genes_monocytes.txt")
genes_name <- "Reynolds, Monocytes"
for (name in names(gene_list)) {
  enrichment.all(genes, genes_name, all_genes_on_450K, gene_list[[name]], name)
}
```

### `Steegenga`

Steegenga, W. T., et. al. ["Genome-wide age-related changes in DNA methylation and gene expression in human PBMCs."](http://link.springer.com/content/pdf/10.1007%2Fs11357-014-9648-x.pdf)

**Genes and methylation changes**

```{r}
genes <- readLines("../data/Steegenga/genes_and_DMR.txt")
genes_name <- "Steegenga, genes and methylation"
for (name in names(gene_list)) {
  enrichment.all(genes, genes_name, all_genes_on_450K, gene_list[[name]], name)
}
```

**Genes, not methylation, changes**

```{r}
genes <- readLines("../data/Steegenga/genes_not_DMR.txt")
genes_name <- "Steegenga, genes not methylation"
for (name in names(gene_list)) {
  enrichment.all(genes, genes_name, all_genes_on_450K, gene_list[[name]], name)
}
```

### `Weidner`

Weidner et al., [“Aging of Blood Can Be Tracked by DNA Methylation Changes at Just Three CpG Sites.”](http://genomebiology.biomedcentral.com/articles/10.1186/gb-2014-15-2-r24) - Beta values for 102 AR-GpGs from 575 samples

```{r}
genes <- readLines("../data/Weidner/genes.txt")
genes_name <- "Weidner"
for (name in names(gene_list)) {
  enrichment.all(genes, genes_name, all_genes_on_450K, gene_list[[name]], name)
}
```


