
```{r setup, echo=FALSE, include=FALSE, cache=FALSE}
# Set up the environment
library(knitr) 
opts_chunk$set(cache.path='cache/', fig.path='img/', cache=F, tidy=T, fig.keep='high', dpi=300, out.width=700, echo=F, autodep=T, width=200)
options(replace.assign=TRUE, width=500)
tidy.opts=list(width.cutoff=200)
suppressMessages(library(pander))
panderOptions('table.split.table', Inf)
set.seed(1)
```

```{r loadData}
trackDb.hg19 <- read.table("data.gr/trackDb.hg19", sep="\t", header=F, row.names=1)
```

```{r loadLibraries, warning=FALSE, message=FALSE}
library(gridExtra)
source("utils.R")
```

Analysis of age-associated genes and genomic regions
========================================================

P-values are corrected for multiple testing using FDR. For each analysis, only the significant enrichment results (p < 0.01) are shown. Negative p-value indicates depletion. Tables show top 10 or less enriched/depleted associations, barplots visualize the same results. If an analysis does not have any significant results, it is not displayed at all.

Hannum meQTLs
---------------

Age-related meQTLs are non-cell type-specifically enriched in CTCF and depleted in Pol2.

```{r Hannum_meQTL_all}
dirs <- list.dirs("data.gr/Hannum", full.names=T, recursive=F) # Paths the SNP sets analyses

for (i in (1:length(dirs))){
  mtx <- read.table(paste(dirs[i], "matrix.txt", sep="/"), sep="\t", row.names=1, header=T)
  colnames(mtx) <- c("All data", "Primary data", "Breast", "Kidney", "Lung")
  if(length(grep("ENCODE", dirs[i])) != 0) {mtx <- mtx[grep("Tfbs", rownames(mtx)), ]}
  mtx <- mtx[, 1, drop=F]
  showTable(mtx)
}
```

Breast- and Kidney-specific age meQTL are enriched in NRSF transcription factor binding cites. 

NRSF = RE1-Silencing Transcription factor (REST), also known as Neuron-Restrictive Silencer Factor (NRSF), is a protein which in humans is encoded by the REST gene, and acts as a transcriptional repressor.[1][2][3] REST is expressly involved in the repression of neural genes in non-neuronal cells.[4][3] Many genetic disorders have been tied to alterations in the REST expression pattern, including colon and small-cell lung carcinomas found with truncated versions of REST.[5] In addition to these cancers, defects in REST have also been attributed a role in Huntington Disease, neuroblastomas, and the effects of epileptic seizures and ischaemia.
Interesting to note that REST strongly correlate with increased longevity. REST levels are highest in the brains of people who lived up to be 90 - 100s and remained cognitively intact.

```{r Hannum_meQTL_tissue}
dirs <- list.dirs("data.gr/Hannum", full.names=T, recursive=F) # Paths the SNP sets analyses

for (i in (1:length(dirs))){
  mtx <- read.table(paste(dirs[i], "matrix.txt", sep="/"), sep="\t", row.names=1, header=T)
  colnames(mtx) <- c("All data", "Primary data", "Breast", "Kidney", "Lung")
  if(length(grep("ENCODE", dirs[i])) != 0) {mtx <- mtx[grep("Tfbs", rownames(mtx)), ]}
  mtx <- mtx[, 3:5, drop=F]
  showTable(mtx)
}
```

Hannum genes
-------------
Genes correlated with age are enriched in Pol2, EZH2-CTCF-RAD21-TAF2 complex. H3K27me3
Polycomb repressive complex 2 is recruited through the interaction of CTCF with Suz12, leading to allele-specific methylation at lysine 27 of histone H3

Enriched in GAD (genomic associations database), conserved. Depleted in genomeicSuperDups.

```{r Hannum_genes}
dirs <- list.dirs("data.gr/Hannum_genes", full.names=T, recursive=F) # Paths the SNP sets analyses

for (i in (1:length(dirs))){
  mtx <- read.table(paste(dirs[i], "matrix.txt", sep="/"), sep="\t", row.names=1, header=T)
  mtx <- mtx[, 1, drop=F]
  rownames(mtx) <- sub("name.", "", rownames(mtx))
  print("---------------------------------------------------------------")
  showTable(mtx)
}
```

Alisch
-------

All enriched in H3K4me1. Methylated are enriched in H3K27me3 (reverse in demethylated). Demethylated are enriched in H3K4me1 (reverse in methylated).
Methylated are enriched in EZH2, SUZ12 (reverse in demethylated, not significant in all).

```{r Alisch}
dirs <- list.dirs("data.gr/Alisch", full.names=T, recursive=F) # Paths the SNP sets analyses

for (i in (1:length(dirs))){
  mtx <- read.table(paste(dirs[i], "matrix.txt", sep="/"), sep="\t", row.names=1, header=T)
  rownames(mtx) <- sub("name.", "", rownames(mtx))
  #mtx <- mtx[, 1, drop=F]
  print("---------------------------------------------------------------")
  showTable(mtx)
}
```

Rakyan
-------

Hypermethylated regions are enriched in EZH2, H3K27me3.

```{r Rakyan}
dirs <- list.dirs("data.gr/Rakyan", full.names=T, recursive=F) # Paths the SNP sets analyses

for (i in (1:length(dirs))){
  mtx <- read.table(paste(dirs[i], "matrix.txt", sep="/"), sep="\t", row.names=1, header=T)
  #mtx <- mtx[, 1, drop=F]
  print("---------------------------------------------------------------")
  showTable(mtx)
}
```